### DISCUSSION TITLE
Join an array in a table display

### DISCUSSION THREAD

**Discussion by diogenesbrussels** [Budibase/budibase#8304] [Help] [answered]
Hello,

I'm trying to join an array in a table display but I'm not able to find a working way to do it.
I have a MySQL query that get the data in the data provider then I use a table component to display it.
One of the columns is an array (from a multiselect component) but when I use {{join Value ","}} or {{join (arrayify Value) ","}} this doesn't work (it displays the array as text). The 'arrayify' function seems to put all the text in one element of a new array (length always equal 1).
Could you help me to resolve this issue?

Thanks!

ps: maybe there should be an 'array' type in the query schema to have this handled properly?

**Comment by andz-bb** [comment, marked as answer]
hey @diogenesbrussels 

I had a look at this and wasn't able to get it working using the bindings because when the column type was array the table would always try to show the value as an array, even if you transform it to be a string in the bindings. As a workaround I was able to do the array join in the transformer of the query, and then set that as a text column in the query schema. 

My transformer looks like this (my array column in the db is MultiSelect): 
`return data.map(x => ({...x, newColumn: x.MultiSelect?.join()}))`

Now the newColumn shows up in my query schema as a text type and will render in the table as you would expect. You can hide the original array column from the table and just use the one built in the transformer going forward.

On your last point that seems totally valid, the server recognises the json column as an array type but doesn't have a value for that in the picker so it show up as null. I'll log an issue for that one 👍 Hope this helps!

**Reply by diogenesbrussels** [reply]
Thanks for the help @andz-bb , it will serve as a solution in the meantime!
If you have the issue number, I'll be happy to follow the progress :)

**Reply by andz-bb** [reply]
hey @diogenesbrussels, I created the issue here: #8389 - thanks again for bringing this to our attention :)
